src/iop/grain.c

```diff
@@ -423,7 +509,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
         noise = _simplex_2d_noise(x + hash, y, octaves, 1.0, zoom);
       }
 
-      out[0] = in[0] + ((100.0 * (noise * (strength))) * GRAIN_LIGHTNESS_STRENGTH_SCALE);
+      out[0] = in[0] + (100.0 * dt_lut_lookup_2d_1c(data->grain_lut, (noise * strength) * GRAIN_LIGHTNESS_STRENGTH_SCALE, in[0] / 100.0));
```

Comment:
Why don't you push the `100.0 *` into the LUT? It's not much, but would save a few cycles. Or would that give different results? I haven't really tried to understand the logic yet.